packages/block-editor/src/hooks/border.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockSupport } from '@wordpress/blocks';
+import { PanelBody } from '@wordpress/components';
+import { Platform } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../components/inspector-controls';
+import { BorderRadiusEdit, useIsBorderRadiusDisabled } from './border-radius';
+
+export const BORDER_SUPPORT_KEY = '__experimentalBorder';
+
+export function BorderPanel( props ) {
+	const isDisabled = useIsBorderDisabled( props );
+	const isSupported = hasBorderSupport( props.name );
+
+	if ( isDisabled || ! isSupported ) {
+		return null;
+	}
+
+	return (
+		<InspectorControls>
+			<PanelBody title={ __( 'Border settings' ) }>
+				<BorderRadiusEdit { ...props } />
+			</PanelBody>
+		</InspectorControls>
+	);
+}
+
+/**
+ * Determine whether there is block support for borders.
+ *
+ * @param {string} blockName Block name.
+ * @return {boolean}         Whether there is support.
+ */
+export function hasBorderSupport( blockName ) {
+	if ( Platform.OS !== 'web' ) {
+		return false;
+	}
+
+	const support = getBlockSupport( blockName, BORDER_SUPPORT_KEY );
+
+	// Further border properties to be added in future iterations.
+	// e.g. support && ( support.radius || support.width || support.style )
+	return true === support || ( support && support.radius );
```

Comment:
```suggestion
	return true === support || support?.radius;
```

If you care about the return type being `boolean`:

```suggestion
	return !! ( true === support || support?.radius );
```